### PR TITLE
Added Localization changes

### PR DESCRIPTION
**Pull request by 4manasa**

### CODE REVIEW

File: src/components/Assignment/index.ts [hunk 1]
```diff
@@ -275,7 +278,8 @@ class Assignment extends BridgeBase {
     this.navigateToStep = actionsAPI.navigateToStep.bind(actionsAPI);
     this.cancelAssignment = actionsAPI.cancelAssignment.bind(actionsAPI);
     this.showPage = actionsAPI.showPage.bind(actionsAPI);
-
+    this.localizedVal = PCore.getLocaleUtils().getLocaleValue;
+    this.localeReference = `${this.thePConn.getCaseInfo().getClassName()}!CASE!${this.thePConn.getCaseInfo().getName()}`.toUpperCase();
```
Comment (niallriddell) [inline src/components/Assignment/index.ts:282]: Is this the only way to get the localization ref?  Would this not already be available (pre-constructed) from the view metadata?

File: src/components/Attachment/index.ts [hunk 2]
```diff
@@ -579,7 +579,7 @@ class Attachment extends BridgeBase {
     for (const file of arFiles) {
       if (!this.validateMaxSize(file, 5)) {
         file.error = true;
-        file.meta = 'File is too big. Max allowed size is 5MB.';
+        file.meta = this.thePConn.getLocalizedValue('File is too big. Max allowed size is 5MB.', '', '');
```
Comment (niallriddell) [inline src/components/Attachment/index.ts:582]: Didn't see a test for this scenario

File: src/components/CancelAlert/index.ts [hunk 3]
```diff
@@ -70,11 +71,12 @@ class CancelAlert extends BridgeBase {
...
       const caseName = caseInfo.getName();
       const ID = caseInfo.getID();
+      this.localizedVal = PCore.getLocaleUtils().getLocaleValue;
 
       this.itemKey = contextName;
       this.heading = `Delete ${caseName} (${ID})`;
-      this.body1 = `Are you sure you want to delete ${caseName} (${ID})?`;
-      this.body2 = 'Alternatively, you can continue working or save your work for later.';
+      this.body1 = `${this.localizedVal('Are you sure you want to delete ', this.localeCategory)} caseName (${ID})?`;
+      this.body2 = this.localizedVal('Alternatively, you can continue working or save your work for later.', this.localeCategory);
```
Comment (niallriddell) [inline src/components/CancelAlert/index.ts:79]: Was there a test for this?

File: src/components/CancelAlert/index.ts [hunk 4]
```diff
@@ -222,7 +227,7 @@ class CancelAlert extends BridgeBase {
             //   timer.unsubscribe();
             //   this.erservice.sendMessage("show", "Save failed.");
             //   });
-            this.sendMessage('Save failed');
+            this.sendMessage(this.localizedVal('Save failed', this.localeCategory));
```
Comment (niallriddell) [inline src/components/CancelAlert/index.ts:230]: Test?

File: src/components/CancelAlert/index.ts [hunk 5]
```diff
@@ -242,7 +247,7 @@ class CancelAlert extends BridgeBase {
             //   timer.unsubscribe();
             //   this.erservice.sendMessage("show", "Delete failed.");
             //   });
-            this.sendMessage('Delete failed.');
+            this.sendMessage(this.localizedVal('Delete failed.', this.localeCategory));
```
Comment (niallriddell) [inline src/components/CancelAlert/index.ts:250]: Test?

File: src/components/ModalViewContainer/index.ts [hunk 6]
```diff
@@ -37,6 +37,8 @@ class ModalViewContainer extends BridgeBase {
   cancelPConn: any;
   bShowCancelAlert = false;
   bAlertState = false;
+  localizedVal: any;
```
Comment (niallriddell) [inline src/components/ModalViewContainer/index.ts:40]: Why any type here?

File: src/components/NavBar/index.ts [hunk 7]
```diff
@@ -26,7 +26,10 @@ class NavBar extends BridgeBase {
   @property({ attribute: false }) portalOperator;
   @property({ attribute: false }) portalOperatorInitials;
   @property({ attribute: false }) navIcon;
-
+  localizedVal: any;
```
Comment (niallriddell) [inline src/components/NavBar/index.ts:29]: Why any type here?  Surely we know it's a string?

File: src/components/NavBar/index.ts [hunk 8]
```diff
@@ -26,7 +26,10 @@ class NavBar extends BridgeBase {
...
+  localeCategory = 'AppShell';
+  localeUtils = PCore.getLocaleUtils();
+  localeReference: any;
```
Comment (niallriddell) [inline src/components/NavBar/index.ts:32]: Why any type here?  Surely we know it's a string?

File: src/components/ToDo/index.ts [hunk 9]
```diff
@@ -74,7 +74,10 @@ class ToDo extends BridgeBase {
 
   bShowMore = true;
   count;
-
+  localizedVal;
```
Comment (niallriddell) [inline src/components/ToDo/index.ts:77]: Implicit any

File: src/components/forms/RadioButtons/index.ts [hunk 10]
```diff
@@ -30,6 +30,12 @@ class RadioButtons extends FormComponentBase {
 
   @property({ attribute: false, type: Array }) options;
 
+  fieldMetadata: any = [];
```
Comment (niallriddell) [inline src/components/forms/RadioButtons/index.ts:33]: Why any type here?

File: src/components/widgets/FileUtility/index.ts [hunk 11]
```diff
@@ -57,18 +57,7 @@ class FileUtility extends BridgeBase {
...
-      id: 'addNewFiles',
-      onClick: () => this.createModal('addLocalFile')
-    },
-    {
-      text: 'Add links',
-      id: 'addNewLinks',
-      onClick: () => this.createModal('addLocalLink')
-    }
-  ];
+  addAttachmentsActions: any[] = [];
```
Comment (niallriddell) [inline src/components/widgets/FileUtility/index.ts:60]: Use of any?

**Review by niallriddell — CHANGES_REQUESTED** [review]
Can you add correct types and also ensure test coverage.  See comments

File: src/components/Attachment/index.ts (hunk 2 above)
Comment (4manasa) [reply src/components/Attachment/index.ts:582]: we need 5MB file to test this use case, which we can't ship with our SDK.

Comment (niallriddell) [reply src/components/Attachment/index.ts:582]: As long as a manual test has been done.

File: src/components/ModalViewContainer/index.ts (hunk 6 above)
Comment (niallriddell) [reply src/components/ModalViewContainer/index.ts:40]: Please use type Function

File: src/components/ToDo/index.ts [hunk 12]
```diff
@@ -74,7 +74,10 @@ class ToDo extends BridgeBase {
 
   bShowMore = true;
   count;
-
+  localizedVal: any;
```
Comment (niallriddell) [inline src/components/ToDo/index.ts:77]: Please use type Function

**Review by niallriddell — APPROVED** [review]
lgtm

**Pull request merged**